storage/tests/unit/test_blob.py

```diff
@@ -20,13 +20,45 @@
 import os
 import tempfile
 import unittest
+import requests
 
 import google.cloud.storage.blob
 import mock
 import six
 from six.moves import http_client
 
 
+def _make_response(status_code=http_client.OK, content=b"", headers={}):
+    response = requests.Response()
+    response.status_code = status_code
+    response._content = content
+    response.headers = headers
+    response.request = requests.Request()
+    return response
+
+
+def _make_connection():
+    from google.cloud.storage._http import Connection
+
+    mock_conn = mock.create_autospec(Connection)
+    mock_conn.user_agent = "testing 1.2.3"
+    mock_conn._make_request.return_value = _make_response(
+        status_code=200,
+        headers={
+            "content-type": "multipart/mixed; boundary=batch_pK7JBAk73-E=_AA5eFwv4m2Q=",
+        },
+        content=b"""
+--batch_pK7JBAk73-E=_AA5eFwv4m2Q=
+HTTP/1.1 404 NotFound
+
+--batch_pK7JBAk73-E=_AA5eFwv4m2Q=
+HTTP/1.1 200 OK
+
+--batch_pK7JBAk73-E=_AA5eFwv4m2Q=--"""
+    )
+    return mock_conn
```

Comment:
Here are two different responses for two different `exists()` calls, and one of them is NotFound